pallets/foreign-investments/src/impls/mod.rs

```diff
@@ -173,10 +173,11 @@ impl<T: Config> ForeignInvestment<T::AccountId> for Pallet<T> {
 		pool_currency: T::CurrencyId,
 	) -> Result<(), DispatchError> {
 		let collected = T::Investment::collect_redemption(who.clone(), investment_id)?;
-		CollectedRedemption::<T>::try_mutate(who, investment_id, |collected_redemption| {
+		CollectedRedemptions::<T>::try_mutate(who, investment_id, |collected_redemption| {
 			collected_redemption
 				.amount_collected
 				.ensure_add_assign(collected.amount_collected)?;
```

Comment:
Why do we store? And for what reasons?

Reply:
By now, we are just storing the amount of tranche tokens which were converted to pool currency. We need to do this as calling `CollectRedeem` and finalizing the collection is asynchronous. So when we want to dispatch `ExecutedCollectRedeem`, we need to have knowledge of that amount.